docs/contribution.rst

```diff
@@ -0,0 +1,125 @@
+======================
+Contributing to ecl2df
+======================
+
+Contributing to ecl2df is easiest on Linux computers. Windows has not been
+tested, and for Mac you will have to compile OPM yourself.
+
+Getting started as a developer
+------------------------------
+
+Clone the master repository from Github::
+
+  git clone https://github.com/equinor/ecl2df
+
+It is recommended that you also fork the repository on Github, in order to push
+your branches to your fork. After forking, add the fork to your cloned copy::
+
+  git remote add fork git@github.com:<youruser>/ecl2df
+
+This requires a valid login setup with SSH keys for you github account, needed
+for write access.
+
+After cloning, you should make a Python virtual environment in which you install
+ecl2df and its dependencies. If you want to create a new for ecl2df, you can
+do something like the following::
+
+  virtualenv venv-ecl2df
+  source venv-ecl2df/bin/activate
+
+and then run ``pip`` ::
+
+  pip install -e .[tests,docs]
+
+to install ecl2df in "edit"-mode together will all dependencies for ecl2df, its
+test suite and documentation.
+
+A good start is to verify that all tests pass after having cloned the
+repository, which you can do by running::
+
+  pytest
+
```

Comment:
pytest tests/ from the cloned directoy.  If pytest does not run from the command line try "python -m pytest tests/"

Reply:
Do we know when we need python -m pytest so we can evaluate how common the error is? 


(There are so many things that can go wrong with every command listed, so catering for all errors is not really rather provide one set of instructions that should work if followed precisely)

Reply:
always pip install pytest in your env first so it overrides any existing (wrong) pytest

Reply:
That follows from `pip install -e .[tests]` above.